### PR TITLE
msm8937-common: treble disaster

### PR DESCRIPTION
**Pull request by JarlPenguin**

### CODE REVIEW

File: rootdir/etc/init.qcom.rc [hunk 1]
```diff
@@ -342,9 +344,9 @@ on post-fs-data
 
     # Mounting of persist is moved to 'on emmc-fs' and 'on fs' sections
     # We chown/chmod /persist again so because mount is run as root + defaults
-    chown root system /persist
-    chmod 0771 /persist
-    chown system system /persist/WCNSS_qcom_wlan_nv.bin
+    chown root system /mnt/vendor/persist
+    chmod 0771 /mnt/vendor/persist
+    chown system system /mnt/vendor/persist/WCNSS_qcom_wlan_nv.bin
```
Comment (wiktorek140) [inline rootdir/etc/init.qcom.rc:348]: Is it used?

Comment (JarlPenguin) [reply rootdir/etc/init.qcom.rc:348]: yes

File: BoardConfigCommon.mk [hunk 2]
```diff
@@ -139,9 +141,7 @@ BOARD_MKBOOTIMG_ARGS := --ramdisk_offset 0x01000000 --tags_offset 0x00000100
 TARGET_KERNEL_ARCH := arm64
 TARGET_KERNEL_HEADER_ARCH := arm64
 TARGET_KERNEL_SOURCE := kernel/motorola/msm8937
-TARGET_KERNEL_CLANG_COMPILE := true
-#KERNEL_TOOLCHAIN := $(PWD)/prebuilts/gcc/$(HOST_OS)-x86/aarch64/aarch64-linux-android-4.9/bin
-#TARGET_KERNEL_CROSS_COMPILE_PREFIX := aarch64-linux-androidkernel-
+TARGET_KERNEL_CROSS_COMPILE_PREFIX := aarch64-linux-androidkernel-
```
Comment (wiktorek140) [inline BoardConfigCommon.mk]: Did not compile with clang?

Comment (JarlPenguin) [reply BoardConfigCommon.mk]: compiled, testing internal error rn

File: BoardConfigCommon.mk [hunk 3]
```diff
@@ -183,7 +192,7 @@ TARGET_PROVIDES_QTI_TELEPHONY_JAR := true
 
 # SELinux
 include device/qcom/sepolicy/sepolicy.mk
-BOARD_SEPOLICY_DIRS += $(COMMON_PATH)/sepolicy
+#BOARD_SEPOLICY_DIRS += $(COMMON_PATH)/sepolicy/vendor
```
Comment (wiktorek140) [inline BoardConfigCommon.mk]: Not used?

Comment (JarlPenguin) [reply BoardConfigCommon.mk]: sepol is disabled

Comment (JarlPenguin) [reply BoardConfigCommon.mk]: for now

File: BoardConfigCommon.mk [hunk 4]
```diff
@@ -193,6 +202,13 @@ PRODUCT_SOONG_NAMESPACES += \
 # Thermal
 USE_DEVICE_SPECIFIC_THERMAL := true
 
+# Treble
+PRODUCT_FULL_TREBLE_OVERRIDE := true
+
+# VNDK
+BOARD_VNDK_VERSION := current
+BOARD_VNDK_RUNTIME_DISABLE := true
```
Comment (wiktorek140) [inline BoardConfigCommon.mk:212]: May i ask what do this param?

Comment (JarlPenguin) [reply BoardConfigCommon.mk:212]: defines vndk version

Comment (JarlPenguin) [reply BoardConfigCommon.mk:212]: disables linker namespace isolation, the 2nd flag

Comment (JarlPenguin) [reply BoardConfigCommon.mk:212]: because our cam blobs don't like it

Comment (JarlPenguin) [reply BoardConfigCommon.mk:212]: ofc need to patch blobs for that later though

File: rootdir/etc/init.qcom.rc (hunk 1 above)
Comment (JarlPenguin) [reply rootdir/etc/init.qcom.rc:348]: need to make bind mount dir in init so

**Review by wiktorek140 — COMMENTED** [review]
Looks good to me, i didn't spot errors

**Comment by JarlPenguin** [comment]
> Looks good to me, i didn't spot errors

ok, let's wait on internal error, then approve

**Review by wiktorek140 — APPROVED** [review]
Approved

**Review by Sohamlad7 — APPROVED** [review]

**Pull request merged**